src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -56,21 +57,30 @@ static zns_sm_metadata_t make_metadata(
   
   auto config_segment_size = get_conf<Option::size_t>(
     "seastore_segment_size");
-  INFO("config segment size: {}", config_segment_size);
-  size_t zones_per_segment = config_segment_size / zone_capacity;
+
+  size_t segment_size = (zone_size == 0) ? config_segment_size : (zone_size << SECT_SHIFT);
+  size_t zones_per_segment = segment_size / (zone_size << SECT_SHIFT);
```

Comment:
should be  size_t zones_per_segment = segment_size / (zone_size == 0 ? config_segment_size : zone_size ); ?

Reply:
Thanks  for the review Liu.  Will fix it.
Also now, I feel that for a ZNS SSD, "zone_size" should never be zero and if at all it was found to be zero then we should probably assert on that before reaching here and assume zone_size will not be zero at this point.

Reply:
Probably add `ceph_assert(zone_size)` right after it is read out from `device.ioctl(BLKGETZONESZ, (void *)&zone_size)`.

Reply:
Yes, I will do that.